site/src/pages/OrganizationSettingsPage/OrganizationSettingsPage.tsx

```diff
@@ -0,0 +1,185 @@
+import type { Interpolation, Theme } from "@emotion/react";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import { useFormik } from "formik";
+import { type FC, useState } from "react";
+import { useMutation, useQueryClient } from "react-query";
+import * as Yup from "yup";
+import {
+  createOrganization,
+  updateOrganization,
+  deleteOrganization,
+} from "api/queries/organizations";
+import type { UpdateOrganizationRequest } from "api/typesGenerated";
+import { ErrorAlert } from "components/Alert/ErrorAlert";
+import {
+  FormFields,
+  FormSection,
+  HorizontalForm,
+  FormFooter,
+} from "components/Form/Form";
+import { IconField } from "components/IconField/IconField";
+import { Margins } from "components/Margins/Margins";
+import { PageHeader, PageHeaderTitle } from "components/PageHeader/PageHeader";
+import { Stack } from "components/Stack/Stack";
+import {
+  getFormHelpers,
+  nameValidator,
+  displayNameValidator,
+  onChangeTrimmed,
+} from "utils/formUtils";
+import { useOrganizationSettings } from "./OrganizationSettingsLayout";
+
+const MAX_DESCRIPTION_CHAR_LIMIT = 128;
+const MAX_DESCRIPTION_MESSAGE =
+  "Please enter a description that is no longer than 128 characters.";
+
+export const getValidationSchema = (): Yup.AnyObjectSchema =>
+  Yup.object({
+    name: nameValidator("Name"),
+    display_name: displayNameValidator("Display name"),
+    description: Yup.string().max(
+      MAX_DESCRIPTION_CHAR_LIMIT,
+      MAX_DESCRIPTION_MESSAGE,
+    ),
+  });
+
+const OrganizationSettingsPage: FC = () => {
+  const queryClient = useQueryClient();
+  const addOrganizationMutation = useMutation(createOrganization(queryClient));
+  const updateOrganizationMutation = useMutation(
+    updateOrganization(queryClient),
+  );
+  const deleteOrganizationMutation = useMutation(
+    deleteOrganization(queryClient),
+  );
+
+  const { currentOrganizationId, organizations } = useOrganizationSettings();
+
+  const org = organizations.find((org) => org.id === currentOrganizationId)!;
+
+  const error =
+    addOrganizationMutation.error ?? deleteOrganizationMutation.error;
+
+  const form = useFormik<UpdateOrganizationRequest>({
+    initialValues: {
+      name: org.name,
+      display_name: org.display_name,
+      description: org.description,
+      icon: org.icon,
+    },
+    validationSchema: getValidationSchema(),
+    onSubmit: (values) =>
+      updateOrganizationMutation.mutateAsync({ orgId: org.id, req: values }),
```

Comment:
I just realized now when using an async function in `onSubmit` we can use the form.isSubmitting 👍